### DISCUSSION TITLE
Where to put robots.txt?

### DISCUSSION THREAD

**Discussion by gphg** [hydephp/hyde#255] [Q&A] [answered]
This is my first week using this static generator. I have a question where should I put robots.txt on project directory so it get copied or rendered (in case it was something like `.blade.php`, YET produced `.txt` file)?

**Comment by emmadesilva** [comment, marked as answer]
Hey! Thanks for using HydePHP! There is no special way to set up robots.txt files using Blade or similar. There is actually no feature dedicated to handling robots.txt files, but maybe there should be?

Depending on how your site is set up, there are three ways you can add a robots file, let me know if you want further information about any of these:
1. Create a [post-build-task](https://hydephp.com/docs/1.x/build-tasks), this can then generate the robots file each time the site is built (for example by copying a file from somewhere else, or just using `file_put_contents()`)
2. If you have your `_site/` directory in version control you can set `empty_output_directory` in `config/hyde.php` to `true`, then Hyde won't empty the `_site/` directory and you can just commit the txt file there
3. If you are using GitHub Actions, you can also add a step to create the file there. We do this on the HydePHP.com website CI for similar things, see https://github.com/hydephp/hydephp.com/blob/master/.github/workflows/build.yml#L72-L73

**Reply by gphg** [reply]
It would be nice if copying static files to root of site is possible. I wish I could do the same on `favicon.ico` too.

**Reply by emmadesilva** [reply]
I fully agree! I haven't figured out the best syntax for it though. Do you have any suggestions? 

So far mine have been:
1. Copying .txt files in the _pages/ directory to _site root, but that feels hacky
2. Or simply not removing these files from the _site/ directory, which could make sense as there is no other way to put them there, but that runs in to trouble when people don't commit the _site/ directory to version control.

But I'm very much open to better idas, as this is something I would like to implement

**Reply by gphg** [reply]
Looking forward for it!

**Comment by tuaris** [comment]
I have a similar requirement with a `BingSiteAuth.xml` file.   I'm just copying it manually to the `_sites` directory.  Doesn't look like HydePHP is blowing away the contents of the directory during build, so it works for now.

**Reply by emmadesilva** [reply]
Thanks for sharing your workflow! It's great for us to know how people use HydePHP. There are a few kinds of files we do clear out, but this can be customized. In general, we recommend that people use BuildTasks for these kinds of things, for example creating or copying custom static files. That makes it truly robust!

**Comment by emmadesilva** [comment]
In the upcoming HydePHP v3 you can do this using the new `_static/` passthrough directory

https://github.com/hydephp/develop/pull/2585
